pkg/sdk/testint/failover_groups_integration_test.go

```diff
@@ -104,8 +138,83 @@ func TestInt_FailoverGroupsCreate(t *testing.T) {
 	})
 }
 
+func TestInt_Issue2544(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	accountName := testenvs.GetOrSkipTest(t, testenvs.BusinessCriticalAccount)
+	businessCriticalAccountId := sdk.NewAccountIdentifierFromFullyQualifiedName(accountName)
+
+	t.Run("alter object types, replication schedule, and allowed integration types at the same time", func(t *testing.T) {
+		id := sdk.RandomAccountObjectIdentifier()
+		objectTypes := []sdk.PluralObjectType{
+			sdk.PluralObjectTypeIntegrations,
+			sdk.PluralObjectTypeDatabases,
+		}
+		allowedAccounts := []sdk.AccountIdentifier{
+			businessCriticalAccountId,
+		}
+		allowedIntegrationTypes := []sdk.IntegrationType{
+			sdk.IntegrationTypeAPIIntegrations,
+			sdk.IntegrationTypeNotificationIntegrations,
+		}
+		replicationSchedule := "10 MINUTE"
+		err := client.FailoverGroups.Create(ctx, id, objectTypes, allowedAccounts, &sdk.CreateFailoverGroupOptions{
+			AllowedDatabases: []sdk.AccountObjectIdentifier{
+				testDb(t).ID(),
+			},
+			AllowedIntegrationTypes: allowedIntegrationTypes,
+			ReplicationSchedule:     sdk.String(replicationSchedule),
+		})
+		require.NoError(t, err)
+		cleanupFailoverGroup := func() {
+			err := client.FailoverGroups.Drop(ctx, id, nil)
+			require.NoError(t, err)
+		}
+		t.Cleanup(cleanupFailoverGroup)
+
+		newObjectTypes := []sdk.PluralObjectType{
+			sdk.PluralObjectTypeIntegrations,
+		}
+		newAllowedIntegrationTypes := []sdk.IntegrationType{
+			sdk.IntegrationTypeAPIIntegrations,
+		}
+		newReplicationSchedule := "20 MINUTE"
+
+		// does not work together:
+		opts := &sdk.AlterSourceFailoverGroupOptions{
+			Set: &sdk.FailoverGroupSet{
+				ObjectTypes:             newObjectTypes,
+				AllowedIntegrationTypes: newAllowedIntegrationTypes,
+				ReplicationSchedule:     sdk.String(newReplicationSchedule),
```

Comment:
If that's not allowed, then maybe it should be moved to sdk.FailoverGroupSet validation (?)

Reply:
Until we get the response about the syntax I would leave like this. If we get the confirmation, we should change the SDK implementation (so like you suggested validation, but also the generation objects, etc)